remote_attestation/rust/src/handshaker.rs

```diff
@@ -505,33 +484,45 @@ impl ServerHandshaker {
         client_identity: ClientIdentity,
         key_negotiator: KeyNegotiator,
     ) -> anyhow::Result<()> {
-        if self.behavior.contains_peer_attestation() {
```

Comment:
I think this logic was incorrect: if the behavior does not have the expected peer attestation, then nothing was being checked. Instead, I think only the attestation report check should be skipped, and in fact even that now is much stricter (if the binary does not expect a remote attestation from the other side, then it will fail if there is one, so that it cannot be accidentally ignored so easily).